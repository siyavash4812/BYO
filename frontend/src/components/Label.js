import styled from "styled-components";
import PropTypes from "prop-types";
import { FontSize } from "../utils/sizes";
import Colours from "../utils/colours";

const Label = styled.span`
  font-size: ${({ size }) => FontSize[size]};
  color: ${Colours.NAVY}; ;
`;

Label.propTypes = {
  size: PropTypes.oneOf(["S", "M", "L"]),
};
Label.defaultProps = {
  size: "S",
};

export default Label;
