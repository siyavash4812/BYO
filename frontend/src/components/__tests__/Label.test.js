import React from "react";
import renderer from "react-test-renderer";

import Label from "../Label";

describe("Label", () => {
  it("renders Label with size 1em if size is not specified", () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-size", "1em");
  });

  it("renders Label with size 1em if size is not valid", () => {
    const tree = renderer.create(<Label />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("font-size", "1em");
  });

  it("renders specified Label", () => {
    const small = renderer.create(<Label size="S" />).toJSON();
    expect(small).toMatchSnapshot();
    expect(small).toHaveStyleRule("font-size", "1em");

    const medium = renderer.create(<Label size="M" />).toJSON();
    expect(medium).toMatchSnapshot();
    expect(medium).toHaveStyleRule("font-size", "1.5em");

    const large = renderer.create(<Label size="L" />).toJSON();
    expect(large).toMatchSnapshot();
    expect(large).toHaveStyleRule("font-size", "2em");
  });
});
