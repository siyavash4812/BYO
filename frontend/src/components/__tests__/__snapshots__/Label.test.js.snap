// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders Label with size 1em if size is not specified 1`] = `
.c0 {
  font-size: 1em;
  color: #001f3f;
}

<span
  className="c0"
  size="S"
/>
`;

exports[`Label renders Label with size 1em if size is not valid 1`] = `
.c0 {
  font-size: 1em;
  color: #001f3f;
}

<span
  className="c0"
  size="S"
/>
`;

exports[`Label renders specified Label 1`] = `
.c0 {
  font-size: 1em;
  color: #001f3f;
}

<span
  className="c0"
  size="S"
/>
`;

exports[`Label renders specified Label 2`] = `
.c0 {
  font-size: 1.5em;
  color: #001f3f;
}

<span
  className="c0"
  size="M"
/>
`;

exports[`Label renders specified Label 3`] = `
.c0 {
  font-size: 2em;
  color: #001f3f;
}

<span
  className="c0"
  size="L"
/>
`;
